app/src/main/java/com/c23ps266/capstoneprojectnew/ui/LoginActivity.kt

```diff
@@ -1,82 +1,56 @@
+                val errorMessage = task.exception?.message
+                Toast.makeText(this@LoginActivity, errorMessage, Toast.LENGTH_SHORT).show()
+            }
+        }
+
         binding.signInButton.setOnClickListener {
-//            signIn()
-            startActivity(Intent(this@LoginActivity, MainActivity::class.java))
-            finish()
+            signInFunction.invoke()
         }
     }
 
-//    private fun signIn() {
-//        val signInIntent = googleSignInClient.signInIntent
-//        resultLauncher.launch(signInIntent)
-//    }
-//
-//    private var resultLauncher: ActivityResultLauncher<Intent> = registerForActivityResult(
-//        ActivityResultContracts.StartActivityForResult()
-//    ) { result ->
-//        if (result.resultCode == Activity.RESULT_OK) {
-//            val task: Task<GoogleSignInAccount> = GoogleSignIn.getSignedInAccountFromIntent(result.data)
-//            try {
-//                // Google Sign In was successful, authenticate with Firebase
-//                val account: GoogleSignInAccount = task.getResult(ApiException::class.java)!!
-//                Log.d(TAG, "firebaseAuthWithGoogle:" + account.id)
-//                firebaseAuthWithGoogle(account.idToken!!)
-//            } catch (e: ApiException) {
-//                // Google Sign In failed, update UI appropriately
-//                Log.w(TAG, "Google sign in failed", e)
-//            }
-//        }
-//    }
-//
-//    private fun firebaseAuthWithGoogle(idToken: String) {
-//        val credential: AuthCredential = GoogleAuthProvider.getCredential(idToken, null)
-//        auth.signInWithCredential(credential)
-//            .addOnCompleteListener(this) { task ->
-//                if (task.isSuccessful) {
-//                    // Sign in success, update UI with the signed-in user's information
-//                    Log.d(TAG, "signInWithCredential:success")
-//                    val user: FirebaseUser? = auth.currentUser
-//                    updateUI(user)
-//                } else {
-//                    // If sign in fails, display a message to the user.
-//                    Log.w(TAG, "signInWithCredential:failure", task.exception)
-//                    updateUI(null)
-//                }
-//            }
-//    }
-//
-//    private fun updateUI(currentUser: FirebaseUser?) {
-//        if (currentUser != null){
-//            startActivity(Intent(this@LoginActivity, MainActivity::class.java))
-//            finish()
-//        }
-//    }
-//
-//    override fun onStart() {
-//        super.onStart()
-//        // Check if user is signed in (non-null) and update UI accordingly.
-//        val currentUser = auth.currentUser
-//        updateUI(currentUser)
-//    }
-//
-//    companion object {
-//        private const val TAG = "LoginActivity"
-//    }
+    override fun onStart() {
+        super.onStart()
+        val currentUser: FirebaseUser? = firebaseAuth.currentUser
+        if (currentUser != null) {
```

Comment:
I think it will be better to check for this user availability from MainActivity.